core/src/dird/job.cc

```diff
@@ -642,16 +642,21 @@ static void* job_thread(void* arg)
   return NULL;
 }
 
-void SdMsgThreadSendSignal(JobControlRecord* jcr, int sig)
+void SdMsgThreadSendSignal_Locked(JobControlRecord* jcr, int sig)
```

Comment:
```suggestion
void SdMsgThreadSendSignal(JobControlRecord* jcr, int sig, const std::unique_lock<std::mutex>& lock)
```

Reply:
we could check the lock is actually held and it is the jcr's mutex, but I'm not sure we really need that:
```
ASSERT(lock);
ASSERT(lock.mutex() == jcr->mutex_guard());
```